### PR TITLE
Convert bootstrap to futures 0.3 + multiple improvements

### PR DESCRIPTION
**Pull request by eugene-babichenko**

### CODE REVIEW

**Comment by mzabaluev** [comment]
This will conflict a lot with #1912, where much of the same is also done.

**Comment by eugene-babichenko** [comment]
Contains a solution for #1830

File: jormungandr/src/main.rs [hunk 1]
```diff
@@ -276,6 +278,8 @@ fn start_services(bootstrapped_node: BootstrappedNode) -> Result<(), start_up::E
         });
     }
 
+    services.spawn_try_future_std("sigint_listener", move |_info| ctrl_c().map_err(|_| ()));
+
```
Comment (NicolasDP) [inline jormungandr/src/main.rs]: this is unfortunate to do that... :/

Comment (eugene-babichenko) [reply jormungandr/src/main.rs]: Actually I can check if can live without that. I was a bit confused by this line from the documentation

> Even if this Signal instance is dropped, subsequent SIGINT deliveries will end up captured by Tokio, and the default platform behavior will NOT be reset.

but Tokio may be able to do some implicit handling.

Comment (eugene-babichenko) [reply jormungandr/src/main.rs]: but it doesn't :(

**Comment by eugene-babichenko** [comment]
Closing this pull request as the changes made here were ported to #1912.

**Pull request closed without merging**